lib/dao.js

```diff
@@ -1643,6 +1644,38 @@ function coerceArray(val) {
   return arrayVal;
 }
 
+DataAccessObject._getHiddenProperties = function() {
+  var settings = this.definition.settings || {};
+  var result = settings.hiddenProperties || settings.hidden;
+  if (typeof result === 'string') {
+    result = [result];
+  }
+  result = result || [];
+  return result;
+};
+
+DataAccessObject._getProtectedProperties = function() {
+  var settings = this.definition.settings || {};
+  var result = settings.protectedProperties || settings.protected;
+  if (typeof result === 'string') {
+    result = [result];
+  }
+  result = result || [];
+  return result;
+};
+
+DataAccessObject._sanitize = function(query, options) {
+  options = options || {};
+
+  // Check violation of keys
+  var prohibitedKeys = this._getHiddenProperties();
+  if (options.excludeProtectedProperties) {
```

Comment:
Should we exclude protected properties by default? I don't fully comprehend what does it mean for a property to be protected and thus whether it should be excluded from queries by default. I am raising this question so that we can consider both alternatives and make the right decision.

Reply:
`protected` properties are hidden from included items but visible to the its own model instances.

Reply:
Makes sense, thanks for the explanation 👍 

Can you please add your explanation as a code comment to help future readers?